src/context/Git.zig

```diff
@@ -0,0 +1,159 @@
+const Git = @This();
+
+const std = @import("std");
+const builtin = @import("builtin");
+const scripty = @import("scripty");
+const context = @import("../context.zig");
+const Signature = @import("doctypes.zig").Signature;
+const Allocator = std.mem.Allocator;
+const DateTime = context.DateTime;
+const String = context.String;
+const Optional = context.Optional;
+const Bool = context.Bool;
+const Value = context.Value;
+
+pub const dot = scripty.defaultDot(Git, Value, false);
+
+pub const gitCommitHashLen = 40;
+
+_in_repo: bool = false,
+
+commit_hash: []const u8 = undefined,
+commit_date: DateTime = undefined,
+commit_message: []const u8 = undefined,
+author_name: []const u8 = undefined,
+author_email: []const u8 = undefined,
+
+@"tag?": ?[]const u8 = null,
```

Comment:
```suggestion
_tag: ?[]const u8 = null,
```

Keep the field secret and create two builtin functions (`tag`, `tag?`) to access it with the same semantics as `git` vs `git?` from my previous comment.


Plase do the same also for `branch?`.